heroku/deploy-a-django-app.md

```diff
@@ -0,0 +1,272 @@
+      - DJANGO_SECRET_KEY=really-super-secret
+```
+
+### Configure Django logging
+
+When Gunicorn is running our app on Heroku, we generally want it to log to stdout
+and stderr instead of logging to files, so that we can let Heroku capture the logs
+and see view them with the `heroku logs` CLI command (or in the web console).
+
+By default, Django will not log errors to the console when `DEBUG` is `False`
+([as documented here](https://docs.djangoproject.com/en/2.2/topics/logging/#django-s-default-logging-configuration)).
+To make sure that errors get logged appropriately in Heroku, set the following
+baseline logging settings in your `settings.py` file:
+
+```python
+import os
+
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,  # Preserve default loggers
+    'handlers': {
+        'console': {
+            'class': 'logging.StreamHandler',
+        },
+    },
+    'loggers': {
+        'django': {
+            'handlers': ['console'],
+            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
+        },
+    },
+}
+```
+
+For more detail on Django's logging framework, [see the documentation](https://docs.djangoproject.com/en/2.2/topics/logging/).
+
+## Provision Heroku resources
+
+Once your application is properly configured for Heroku, the following instructions
+will help you deploy your application to the platform.
+
+### Ensure that the Heroku CLI is installed
+
+The fastest way to get a project up and running on Heroku is to use the
+[Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Before you start,
+make sure you have the CLI installed locally. In addition, confirm that you have the
+manifest plugin) installed:
+
+```
+heroku plugins | grep manifest
+```
+
+If you don't see any output, [follow the official instructions for installing the
+plugin](https://devcenter.heroku.com/changelog-items/1441).
+
+### Create Heroku config files
+
+Define config files relative to the root of your repo.
+
+#### `heroku.yml`
+
+The `heroku.yml` manifest file tells Heroku how to build and network the services
+and containers that comprise your application. This file should live in the root of your project repo.
+For background and syntax, see [the
+documentation](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml).
+Use the following baseline to get started:
+
+```yml
+# Define addons that you need for  your project, such as Postgres, Redis, or Solr.
+setup:
+  addons:
+    - plan: heroku-postgresql
+# Define your application's Docker containers.
+build:
+  docker:
+    web: Dockerfile
+# Define any scripts that you'd like to run every time the app deploys.
+release:
+  command:
+    - ./scripts/release.sh
+  image: web
+# The command that runs your application. Replace 'app' with the name of your app.
+run:
+  web: gunicorn -t 180 --log-level debug app.wsgi:application
+```
+
+#### `release.sh`
+
+In your app's `scripts` folder, define a script `release.sh` to run every time the app deploys.
```

Comment:
Release logs don't steam by default. If your logs aren't helpful, e.g., empty release logs, try the main application log: `heroku logs -a <YOUR APP>`. If you're getting permissions errors on your release script, `chmod u+x scripts/release.sh` locally, then commit and push the updated script.

Reply:
Maybe this would be better suited for the section on debugging!